File: AppSistemaTienda/src/app/Interfaces/venta.ts
import { DetalleVenta } from "./detalle-venta"

export interface Venta {
    idVenta?:number,
    numeroDocumento?:string,
    pago:string,
    totalTexto:string,
    fechaRegistro?:string,
    detalleVenta: DetalleVenta[]
}

File: APISistemaTienda/APISistemaTienda/Models/Venta.cs
using System;
using System.Collections.Generic;

namespace APISistemaTienda.Models
{
    public partial class Venta
    {
        
        public int IdVenta { get; set; }
        public string? NumeroDocumento { get; set; }
        public string? Pago { get; set; }
        public decimal? Total { get; set; }
        public DateTime? FechaRegistro { get; set; }

        public virtual ICollection<DetalleVenta> DetalleVenta { get; } = new List<DetalleVenta>();
    }
}
